assets/src/data/model/entity-factory/constants.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External imports
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { isUndefined } from 'lodash';
+
+/**
+ * Constants describing the current "save state" for an entity.
+ *
+ * @type {{CLEAN: Symbol, NEW: Symbol, DIRTY: Symbol}}
+ */
+export const SAVE_STATE = {
+	CLEAN: Symbol( 'Entity is persisted.' ),
+	NEW: Symbol( 'Entity is new.' ),
+	DIRTY: Symbol( 'Existing entity has changes and needs persisted.' ),
```

Comment:
I don't think it should be a blanket behaviour.  In _most_ cases it should be fine, but we have to be wary of places where Symbols are used and referenced because there are some quirks with how Symbols are transpiled for older browsers.  See [here](https://github.com/zloirock/core-js#caveats-when-using-symbol-polyfill) for more info